rotkehlchen/chain/ethereum/modules/ens/decoder.py

```diff
@@ -153,11 +166,108 @@ def _decode_name_renewed(
+                manager=self.ethereum_manager,
+                method_name='name',
+                arguments=[node],
+            )
+            if ens_name == '':
+                # By checking the contract code, I don't think it can happen. But just in case.
+                return None, []
+
+            # Not able to give more info to the user such as address that was set since
+            # we don't have historical info and event doesn't provide it
+            notes = f'Set ENS address for {ens_name}'
+            decoded_events.append(HistoryBaseEntry(
+                event_identifier=transaction.tx_hash,
+                sequence_index=self.base.get_sequence_index(tx_log),
+                timestamp=ts_sec_to_ms(transaction.timestamp),
+                location=Location.BLOCKCHAIN,
+                location_label=transaction.from_address,
+                asset=A_ETH,
+                balance=Balance(),
+                notes=notes,
+                event_type=HistoryEventType.INFORMATIONAL,
+                event_subtype=HistoryEventSubType.NONE,
+                counterparty=CPT_ENS,
+            ))
+        return None, []
+
+    def _decode_ens_public_resolver_2_events(
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[List[ActionItem]],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], List[ActionItem]]:
+        if tx_log.topics[0] == TEXT_CHANGED:
+            try:
+                _, decoded_data = decode_event_data_abi_str(tx_log, TEXT_CHANGED_ABI)
+                changed_key = decoded_data[0]
+            except (DeserializationError, KeyError) as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'missing key entry for {msg}'
+
+                log.debug(f'Failed to decode ENS set-text event due to {msg}')
+                return None, []
+
+            node = tx_log.topics[1]
+            address = ENS_PUBLIC_RESOLVER_2.call(
+                manager=self.ethereum_manager,
+                method_name='addr',
+                arguments=[node],
+            )
+            address = to_checksum_address(address)
+            try:
+                ens_mapping = find_ens_mappings(
+                    ethereum_manager=self.ethereum_manager,
+                    addresses=[address],
+                    ignore_cache=False,
+                )
+            except RemoteError as e:
+                log.debug(f'Failed to decode ENS set-text event due to {str(e)}')
+                return None, []
+
+            if address in ens_mapping:
+                name_to_show = ens_mapping[address]
+            else:
+                name_to_show = address
+
+            notes = f'Set ENS {changed_key} for {name_to_show}'
```

Comment:
```suggestion
            notes = f'Set ENS {changed_key} attribute for {name_to_show}'
```

Reply:
Also can you know from the transaction which one? That is info I would like to see when reading at the event

Reply:
Sorry what do you mean by `which one` ?

Reply:
so for example if I'm changing twitter, discord telegram...

Reply:
This is contained in `changed_key`

Reply:
Ah okey. Perfect then